### PR TITLE
Initial Upstream Commit form Downstream openshift/cluster-nfd-operator

### PR DESCRIPTION
**Pull request by zvonkok**
Initial Upstream Commit form Downstream openshift/cluster-nfd-operator

### CODE REVIEW

**Comment by zvonkok** [comment]
@marquiz can you add me to the kubernetes_incubator repository on quay so that I can push an initial version of the operator and the e2e tests can be verified? Thanks.

**Comment by zvonkok** [comment]
Can we run any CI on this repository?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM registry.access.redhat.com/ubi8/go-toolset AS builder
+WORKDIR /go/src/github.com/kubernetes-sigs/node-feature-discovery-operator
+COPY . .
+RUN make build
+
+FROM registry.access.redhat.com/ubi8/ubi
+ARG CSV=4.2
```
Comment (zvonkok) [inline Dockerfile]:
```suggestion

```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+FROM registry.access.redhat.com/ubi8/ubi
+COPY --from=builder /go/src/github.com/kubernetes-sigs/node-feature-discovery-operator/node-feature-discovery-operator /usr/bin/
+
+RUN mkdir -p /etc/kubernetes/node-feature-discovery/assets
+COPY assets/ /etc/kubernetes/node-feature-discovery/assets
+
+#ADD controller-manifests /manifests
+
+COPY deploy/olm-catalog/$CSV /manifests/$CSV
```
Comment (zvonkok) [inline Dockerfile]:
```suggestion

```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+FROM registry.access.redhat.com/ubi8/ubi
+COPY --from=builder /go/src/github.com/kubernetes-sigs/node-feature-discovery-operator/node-feature-discovery-operator /usr/bin/
+
+RUN mkdir -p /etc/kubernetes/node-feature-discovery/assets
+COPY assets/ /etc/kubernetes/node-feature-discovery/assets
+
+#ADD controller-manifests /manifests
+
+COPY deploy/olm-catalog/nfd.package.yaml /manifests/
```
Comment (zvonkok) [inline Dockerfile]:
```suggestion
```

**Comment by marquiz** [comment]
> can you add me to the kubernetes_incubator repository on quay so that I can push an initial version of the operator and the e2e tests can be verified? Thanks.

I created a new container image repository
https://quay.io/repository/kubernetes_incubator/node-feature-discovery-operator

I added you as a maintainer of that repo

**Comment by marquiz** [comment]
> Can we run any CI on this repository?

I added notification to travis-ci.org

**Pull request closed without merging**